tests/test_lora_fsdp.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pytest
+import torch
+import torch.nn as nn
+from torch.distributed import destroy_process_group, init_process_group
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from peft import LoraConfig, TaskType, get_peft_model
+from peft.tuners.tuners_utils import BaseTunerLayer
+from peft.utils import ModulesToSaveWrapper
+
+
+# Seeding for deterministic behavior
+torch.manual_seed(42)
+torch.cuda.manual_seed_all(42)
+
+
+@pytest.fixture(scope="module")
+def setup_dist():
+    """Setup and teardown a single-GPU FSDP process group."""
+    if not torch.cuda.is_available():
+        pytest.skip("CUDA required for FSDP tests")
+
+    os.environ["MASTER_ADDR"] = "localhost"
+    os.environ["MASTER_PORT"] = "12355"
+
+    if not torch.distributed.is_initialized():
+        init_process_group(backend="nccl", rank=0, world_size=1)
+
+    yield
+
+    if torch.distributed.is_initialized():
+        destroy_process_group()
+
+
+@pytest.fixture(scope="module")
+def tokenizer():
+    """Reusable tokenizer for consistency across tests."""
+    tok = AutoTokenizer.from_pretrained("gpt2")
+    tok.pad_token = tok.eos_token
+    return tok
+
+
+def ensure_lora_initialized(model):
+    """Ensure LoRA adapter weights are non-zero for stable testing."""
+    for _, module in model.named_modules():
+        if hasattr(module, "lora_A") and hasattr(module, "lora_B"):
+            for adapter_name in module.lora_A:
+                lora_A = module.lora_A[adapter_name]
+                lora_B = module.lora_B[adapter_name]
+                if getattr(lora_A, "weight", None) is not None and lora_A.weight.std() < 1e-6:
+                    nn.init.kaiming_uniform_(lora_A.weight, a=1)
+                if getattr(lora_B, "weight", None) is not None and lora_B.weight.std() < 1e-6:
+                    nn.init.kaiming_uniform_(lora_B.weight, a=1)
+
+
+class TestLoraFSDP:
```

Comment:
When I revert the changes in `set_adapter_layers`, the tests still pass for me (testing on a machine with 2 CUDA GPUs). Is it the same for you?